src/udiskslinuxdrive.c

```diff
@@ -445,6 +445,33 @@ ptr_str_array_compare (const gchar **a,
   return g_strcmp0 (*a, *b);
 }
 
+static gboolean
+find_removable (const GUdevDevice *device)
+{
+  GUdevDevice *parent;
+
+  parent = g_object_ref (device);
+  while (parent)
+  {
+	  GUdevDevice *d;
+	  const gchar **removable;
+
+	  removable = g_udev_device_get_sysfs_attr_as_strv (parent, "removable");
```

Comment:
Because it's a string instead of a boolean, i.e. `ATTRS{removable}=="removable"`.

Reply:
Yeah, but look at the line 517 in your patch - it's an old code that retrieves the value as boolean. I guess it's just a trick treating non-empty string as TRUE and it was working for ages.

Reply:
Because it's really boolean. The sysfs attribute is created based on `GENHD_FL_REMOVABLE` flag. CD-ROM or USB drive have this.

OTOH, the NVMe external drive's `removable` is created based on `DEVICE_REMOVABLE` and it's a string.

Reply:
So is it enough to get this merged?

Reply:
Ahh, I see now. Didn't expect that.